src/components/admin/TagModal.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import map from 'lodash/map';
+import forEach from 'lodash/forEach';
+import {Modal, Button} from 'react-bootstrap';
+import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
+import update from 'immutability-helper';
+
+import config from '../../config';
+
+class TagModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.submitForm = this.submitForm.bind(this);
+    this.state = {
+      tag: {
+        label: {}
+      },
+      labelLanguages: this.constructor.initializeLanguages()
+    };
+  }
+
+  static initializeLanguages() {
+    const labelLanguages = {};
+    forEach(config.languages, (language) => {
+      labelLanguages[language] = false;
+    });
+    return labelLanguages;
+  }
+
+  componentWillMount() {
+    this.setState(update(this.state, { labelLanguages: { fi: { $set: true }}}));
+  }
+
+  onLabelChange(language, value) {
+    this.setState(update(this.state, {
+      tag: {
+        label: {
+          [language]: {
+            $set: value
+          }
+        }
+      }
+    }));
+  }
+
+  onActiveLanguageChange(language) {
+    this.setState(update(this.state, {
+      labelLanguages: {
+        [language]: {
+          $set: !this.state.labelLanguages[language]
+        }
+      }
+    }));
+  }
+
+  submitForm(event) {
+    event.preventDefault();
+    this.props.onCreateTag(this.state.tag);
+    this.props.close();
+  }
+
+  generateCheckBoxes() {
+    const checkBoxes = map(config.languages, (language) => (
+      <div key={language} className={'checkbox-container'}>
+        <FormattedMessage id={`inLanguage-${language}`}/>
+        <input
+          type="checkbox"
+          checked={this.state.labelLanguages[language]}
+          onChange={() => this.onActiveLanguageChange(language)}
+        />
+      </div>
+    ));
+    return <div className="label-checkboxes">{checkBoxes}</div>;
+  }
+
+  generateLabelInputs(intl) {
```

Comment:
Why is `intl` passed into this function? It uses `this.state` anyway, it could just as well snarf `intl` from `this.props`...